third-party/libfabric/Makefile

```diff
@@ -0,0 +1,69 @@
+ifndef CHPL_MAKE_HOME
+export CHPL_MAKE_HOME=$(shell pwd)/../..
+endif
+
+CHPL_MAKE_HOST_TARGET = --target
+include $(CHPL_MAKE_HOME)/make/Makefile.base
+
+#
+# Cray X* builds are cross-compilations.
+#
+ifneq (, $(filter cray-x% cray-shasta,$(CHPL_MAKE_TARGET_PLATFORM)))
+CHPL_LIBFABRIC_CFG_OPTIONS += --host=$(shell uname -m)-cle-linux-gnu
+else ifneq ($(CHPL_MAKE_HOST_PLATFORM),$(CHPL_MAKE_TARGET_PLATFORM))
+CHPL_LIBFABRIC_CFG_OPTIONS += --host=$(CHPL_MAKE_TARGET_PLATFORM)-unknown-linux-gnu
+endif
+
+CHPL_LIBFABRIC_CFG_OPTIONS += --enable-static \
+                              --disable-shared \
+                              --prefix=$(LIBFABRIC_INSTALL_DIR)
+
+ifdef CHPL_COMM_DEBUG
+CHPL_LIBFABRIC_CFG_OPTIONS += --enable-debug
+CFLAGS += -g
+endif
+
+ifeq ($(CHPL_LIB_PIC),pic)
+CFLAGS += $(SHARED_LIB_CFLAGS)
+endif
+
+CHPL_LIBFABRIC_CFG_OPTIONS += $(CHPL_LIBFABRIC_MORE_CFG_OPTIONS)
+
+default: all
+
+all: libfabric
+
+clean: FORCE
+	rm -rf $(LIBFABRIC_BUILD_SUBDIR)
+
+cleanall: FORCE
+	rm -rf build
+
+clobber: FORCE
+	rm -rf build install
+
+libfabric: configure-libfabric build-libfabric install-libfabric
+
+configure-libfabric: FORCE
+#
+# These first few lines touch some autoconf-oriented files in a
+# certain order to prevent autoconf from running again.
+#
+	touch $(LIBFABRIC_SUBDIR)/configure
+	find $(LIBFABRIC_SUBDIR)/. -name Makefile.in | xargs touch
```

Comment:
Probably need the sleeps in between touching files like we have in the other third-party makefiles. Does it make sense to just copy what's in hwloc/qthreads wholesale, or is the libfabric set up different?

Reply:
I did copy  from there, although I adapted somewhat. Those 2 lines are effectively the same as the last 2 done in the hwloc and qthreads Makefiles. In those there is a sleep-delay in front of them that is only there to separate them from other touches done just before these. Here we don't need those previous touches, so we don't need the sleep-delays either.